src/core/utils/poller.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, timer } from 'rxjs';
+
+/**
+ * Create an Observable BehaviorSubject to invoke a function on an interval
+ * which returns the next value for the observable.
+ * @public
+ */
+export class Poller<T> {
+  /**
+   * The observable to observe for changes to the poller value.
+   */
+  public readonly subject$ = new BehaviorSubject<T>(this.initialValue);
+  private poller$ = timer(0, this.frequency);
+  private subscription = this.poller$.subscribe(async iteration => {
+    const next = await this.handler(iteration);
```

Comment:
`subject$` value can have a race condition, because handler is async.

Reply:
This would be the responsibility of the consumer to avoid race conditions. We cannot say for certain that consumers would want to prevent the next poll because their current handler has not completed.